### DISCUSSION TITLE
multiple duplicate product images

### DISCUSSION THREAD

**Discussion by mohitkumar29** [swellstores/community#525] [Help] [answered]
hey there, we are facing the issue where some products have multiple duplicate images in the data returned from swell API, I checked it in the swell dashboard console as well. The product i am attacing screenshot of only has 9 images, but in the API reponse it has 45 images (swell dashboard console screenshot attached).

![image](https://github.com/swellstores/community/assets/104753963/11d9b84b-78a0-424d-a921-8984ba2be5db)

![image](https://github.com/swellstores/community/assets/104753963/e4bb4671-6a05-4e05-8eae-153906ef2007)

**Comment by logeshswell** [comment]
1. Did you try uploading these images via Swell admin dashboard?
2. Did you upload the same image more than once?

**Reply by mohitkumar29** [reply]
as confirmed by the content editor, when changes/updates were needed the existing screenshots were deleted first and then upload the updated ones, shouldn't it be in sync with what is being shown in the admin dashboard? in the UI we see 9 images but 45 images in the API response.

**Reply by logeshswell** [reply]
> when changes/updates were needed

What kind of changes do you mean?

**Reply by logeshswell** [reply]
> shouldn't it be in sync with what is being shown in the admin dashboard?

Yes, but in the Swell admin dashboard, we hide the duplicates and render the unique images.

**Reply by logeshswell** [reply]
Can you also share the screen (or loom) recording of the issue with the steps to reproduce this issue?

**Reply by mohitkumar29** [reply]
> What kind of changes do you mean?

   The changes in the screenshots, the content editor may have re-uploaded the same image after deleting it.
   
   
> Yes, but in the Swell admin dashboard, we hide the duplicates and render the unique images. 

  so as i understand, swell is not actually deleting the screenshots from the database when you delete them? but why? 
  
  
> Can you also share the screen (or loom) recording of the issue with the steps to reproduce this issue?
 
  I am not sure what to record here, we are simply fetching the product using swell-js lib 
  
  ` const product = await swell.products.get(productSlug, {
      expand: ['categories', 'up_sells.product']
    });` 
    
 to reproduce the issue, you can upload an image, delete it, upload it again then try fetching the product and see that it has duplicate 
 images, which it shouldn't have.

**Reply by logeshswell** [reply]
> swell is not actually deleting the screenshots from the database when you delete them?

No, Swell deletes the screenshots when you save the changes.

**Reply by logeshswell** [reply]
I could reproduce this issue and corrected it by manually deleting the duplicate images in the given product.

**Reply by logeshswell** [reply]
The issue occurs when you edit the product image and click the Save button more than once.
If you click the **Save** button 5 times, then 5 duplicate images are created.

It is pretty common to experience this issue with a bad internet connection or larger image file size because the editor modal closes only after the image is successfully uploaded.

**Reply by logeshswell** [reply, marked as answer]
I will raise a bug ticket and will address this issue for the above conditions. I will get back to you if I get some updates.
